python/py-faiss/Portfile

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
```

Comment:
I think this shouldn't be a separate port - compiling the Python interface should in my opinion be part of the ```libfaiss``` port (likely using variants as is done for many other ports).

Reply:
@reneeotten Would you please point to a candidate Portfile as a reference for this? I'm not familiar with how this is done elsewhere.

Reply:
@reneeotten Would you please point to a reference Portfile that does this? I’m not familiar how this is done outside the `python/*` ports.

Reply:
I think one example that makes py-subports "manually" is `lalinference`. It might make sense to combine this - especially since much what you put in the `py-faiss` Portfile (e.g., compiler selection, configure flags, etc...) should likely be the same as for the `libfaiss` port.

Reply:
Looking at https://github.com/macports/macports-ports/blob/master/science/lalinference/Portfile doing this manually is an enormous amount of duplicative work to replicate what should be the default.

Wouldn’t make more sense to create a py-faiss port, include a library-install-only feature as a subport, and simply deprecate math/libfaiss?

Reply:
I am not sure if it's and "enormous amount of duplicative work", probably not more than adding a new `py-faiss` port, but it's fine with me. If you want to go with that options, then the library-install-only port should remain the `math/libfaiss` port, and you add then a new `py-faiss` port with the Python interface; as @mf2k answered on your e-mail to `macports-devel`.

Reply:
Two ports would involve syncing at update.

I’ll just copy all the necessary python tcl subroutines from `/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/python-1.0.tcl`.

Is there some way to use with portgroup in a subport of `libfaiss` to automatically get the names `py37-faiss`? Or must the original stub name be `py-faiss`?

Reply:
you can do that and see what other people think of that idea. I still think that just doing something like
```
foreach v {27 36 37 38} {
  subport py${v}-${name} {
    set pythonversion     [string range ${v} 0 end-1].[string index ${v} end]
    set pythonhome        "${frameworks_dir}/Python.framework/Versions/${pythonversion}"
    set pythondir             "${pythonhome}/lib/python${pythonversion}/site-packages"

    <whatever else is needed>
```
is the way to go (simple enough, and very clear what is intended) instead of copying `tcl` procedures/stuff from a PG into a Portfile...

Reply:
Thanks. Superseded by https://github.com/macports/macports-ports/pull/7514 with your suggestions.